tools/rosgraph/src/rosgraph/impl/graph.py

```diff
@@ -200,9 +203,9 @@ def __init__(self, start, end, label=''):
         self.start = start
         self.end = end
         self.label = label
-        self.key = "%s|%s"%(self.start, self.label)
+        self.key = (self.start, self.label)
```

Comment:
No, `|` is not a valid character in node or edge names. So it is safe to be used as a separator.

Yes, please update the PR - and squash the commits on this branch. The `__contains__` is certainly the significant of this PR. The slightly easier implementation of `__iter__` as well as the set initializations below are more cosmetic but you can keep those.